### PR TITLE
Introduce docker pruning to all extensions / tests that use docker

### PR DESCRIPTION
**Pull request by geoand**
This is done to address various weird issues we are seeing in CI lately.

### CODE REVIEW

**Comment by gsmet** [comment]
What does it do exactly? Especially wondering what's happening for the other Docker containers unrelated to Quarkus you might have?

**Comment by geoand** [comment]
Here are the explanations:

`docker container prune`:
```
Remove all stopped containers
```

`docker image prune`
```
Remove unused images
```

`docker network prune`
```
Remove all unused networks
```

`docker volume prune`
```
Remove all unused local volumes
```

The only one that we could argue about here is `docker container prune`. But I believe we need that in order to free up space in CI when a ton of containers are being running (in JVM tests for example).

**Comment by geoand** [comment]
I understand the concern, you are worried that someone running our tests on their machine will get their containers removed. Right?

**Comment by geoand** [comment]
@gsmet How about the update I just made?

I made it conditional on the build being run inside Github Actions.

**Comment by cescoffier** [comment]
Yes, don't run it on the user machine :-) 
That should mitigate the space issue we are starting to see. 

I was wondering if we could also reduce the history size and checkout only the last commit instead of the whole history.

**Comment by geoand** [comment]
@cescoffier we can certainly do that as well. That's a great idea

**Comment by geoand** [comment]
I need to check however, maybe github does that automatically.

**Comment by geoand** [comment]
Seems like checkout only fetches one commit by default: https://github.com/actions/checkout#whats-new

File: docker-prune.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+# pruning is only run when inside CI to avoid accidentally removing stuff
+# GITHUB_ACTIONS is always set to true inside Github Actions
+# https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables
+if [ "${GITHUB_ACTIONS}" == true ] ; then
+  docker container prune -f
+  docker image prune -f
+  docker network prune -f
+  docker volume prune -f
+fi
```
Comment (gsmet) [inline docker-prune.sh:13]: Let's move it to the .github directory then?

Comment (geoand) [reply docker-prune.sh]: Good idea. Fixed

**Review by gsmet — APPROVED** [review]
Looks good to me.

Another thing that I'm wondering is if snapshots could be piling up in the cache.

**Comment by gsmet** [comment]
Btw, maybe we need a du -sh at the end of the build just to check what's going on. In a couple of iterations, we should understand what's doing on.

**Comment by geoand** [comment]
For the cron job build you mean?

**Comment by gsmet** [comment]
Yes, we are shooting in the dark right now. I was wondering if we should try to analyze what's going on instead.

**Comment by geoand** [comment]
Let's first see if this prunning thing have any effect :)

**Pull request merged**